components/aws/sagemaker/common/_utils.py

```diff
@@ -517,7 +559,17 @@ def create_endpoint_config_request(args):
 
     return request
 
+def get_endpoint_config_name(client, args):
+    endpoint_config_name = args['endpoint_config_name']
+    ## boto3 documentation says to update an endpoint, a new EndPointConfig has to be created
+    ## and the one currently in use should NOT be deleted. Appending a random number to resolve conflict
+    if args['update_endpoint'] and endpoint_config_name_exists(client, endpoint_config_name):
+        endpoint_config_name = endpoint_config_name + "-" + id_generator(8)
+        logging.info("Changed endpoint_config_name to: " + endpoint_config_name)
+    return endpoint_config_name
+
 def create_endpoint_config(client, args):
+    args['endpoint_config_name'] = get_endpoint_config_name(client, args)
```

Comment:
just noticed we have two places where we are trying to generate endpoint_config_name, one is line 572 in this PR and other is [here](https://github.com/kubeflow/pipelines/blob/master/components/aws/sagemaker/common/_utils.py#L488)

Can we remove one/merge these? Also need to handle the case where user hasnt passed the endpoint config name (we dont want to call endpoint_config_name_exists() for a None value)

Reply:
also looks like older implementation - https://github.com/kubeflow/pipelines/blob/master/components/aws/sagemaker/common/_utils.py#L488 is using model name to generate endpoint config which does not guarantee it will generate a uniquename. Your method is better

Reply:
we need the args to check if --update_endpoint is true. I merged the two implementations to use randomly generated names